readme.md

```diff
@@ -74,8 +74,9 @@ where the glob starts, so for example for a  `gulp.src('/base/path/**')` it woul
 * **baseFile** - *string* : Path to the file you wanna pull the subdirectories name from. The Path must be relative
 to the subdirectory that's to be renamed (so the first directory after the base-path). It's content will be given as a
 paramter to `renameTo()`.
-* **renameTo** - *function(baseFileContent)* : The functions return value will be used as the new subdirectories name.
-It's parameter are the contents of the specified `baseFile` (as a string).
+* **renameTo** - *function(baseFileContent, oldSubDirName)* : The functions return value will be used as the new subdirectories name.
+It's first parameter are the contents of the specified `baseFile` (as a string) and second parameter is `old name of directory` that will be changed. If `""`or `undefined` or `null` is returned from the function then no rename is done and folder is copied as it is.
+* **isOptional** - *boolean* : When `baseFile` is not found in folder, processing checks for `isOptional` flag and if its `false` then it throws error otherwise it continues to process with old name. In case of `isOptional = true` and no `baseFile` present in folder, `renameTo` function will receive `baseFileContent` as `undefined` or `null`, in such case implementor can use second argument, that is old name and process on it to generate and return new name to be used for the folder. When `isOpional = true` and there are files in base folder then those files will be copied as it is. Default value is `false`.
```

Comment:
Yea I agree it's quite big but only reason I thought it would makes senses
because repo does not have wiki and this is only place to describe. But
yeah, if you feel like it's big; please feel free to cut it and merge.

Thanks

On Sunday 24 July 2016, David Losert <notifications@github.com> wrote:

> In readme.md
> <https://github.com/davelosert/gulp-subdir-rename/pull/2#discussion_r71986202>
> :
>
> > @@ -74,8 +74,9 @@ where the glob starts, so for example for a  `gulp.src('/base/path/**')` it woul
> >  * **baseFile** - *string* : Path to the file you wanna pull the subdirectories name from. The Path must be relative
> >  to the subdirectory that's to be renamed (so the first directory after the base-path). It's content will be given as a
> >  paramter to `renameTo()`.
> > -* **renameTo** - *function(baseFileContent)* : The functions return value will be used as the new subdirectories name.
> > -It's parameter are the contents of the specified `baseFile` (as a string).
> > +* **renameTo** - *function(baseFileContent, oldSubDirName)* : The functions return value will be used as the new subdirectories name.
> > +It's first parameter are the contents of the specified `baseFile` (as a string) and second parameter is `old name of directory` that will be changed. If `""`or `undefined` or `null` is returned from the function then no rename is done and folder is copied as it is.
> > +* **isOptional** - *boolean* : When `baseFile` is not found in folder, processing checks for `isOptional` flag and if its `false` then it throws error otherwise it continues to process with old name. In case of `isOptional = true` and no `baseFile` present in folder, `renameTo` function will receive `baseFileContent` as `undefined` or `null`, in such case implementor can use second argument, that is old name and process on it to generate and return new name to be used for the folder. When `isOpional = true` and there are files in base folder then those files will be copied as it is. Default value is `false`.
>
> This is a little bit long imho. Maybe we could change it to something like:
>
>    - *isOptional* - *boolean (default: false)* : set this to true if you
>    have folders which do not contain the baseFile. Per default, those folders
>    would cause an error. If set to true, those folders will then just be
>    copied as they are with their original dir-name.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/davelosert/gulp-subdir-rename/pull/2/files/51bc85aa95c4c9c7bf78eb90c3735040741104e7#r71986202>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AAnkeL8CFJpygW5c7rDlGVMsugJx-QgYks5qYz9DgaJpZM4JJRfH>
> .
>